### PR TITLE
Notify users when there is new version of ANET or ANET server is down

### LINKED ISSUE #1164: Add notification banner message when network goes offline
While working on ANET. When the network connection fails there is no indication that the server cannot be reached. It seems like ANET has stopped working properly. Some pages continue to be viewable, however searches and other async calls stop working. This causes frustration with ANET, while the errors are caused by network issues.

Additional point of frustration is when a user leaves ANET open in a browser overnight and we upgrade ANET at night. The user comes in the morning and attempts to use ANET without refreshing the browser which results in errors.

Also consider adding:
`onlineUsers` - returns the currently/recently logged users, last time of service use and ip address (or an approximation - i.e. the most recet 20 users)

**Proposed Solution**
Design an functionality that detects new versions and drops in connectivuty, indicates it to the user, and possibly prevents actions that result in data loss

### LINKED ISSUE #3020: Add notification banner message when there is a new version of ANET
When a new version of ANET is published, it is problematic for users that leave their browsers open overnight, they end up with a version out of sync.
Have ANET check for new versions, and when the case, automatically reload the client if the page contains no unsaved data. 
Add notification banner message when not possible

related to #3019 and #1164

### PR DESCRIPTION
**Pull request by cemalettin-simsoft**
I have implemented a polling mechanism from client-side for version data, if there is a new one, we notify new version info, if there is a polling error, we notify server/network down info.

The banner text at the top will be changed when a new version of ANET detected or server being down detected. Users will see a different banner text with different background color. This way, they can make sure they are on the latest version of ANET and they will not do any data losing action when server is down as well.

Closes #1164, closes #3020 

#### User changes
- A new ANET version and ANET server being down notifications under the top security banner

#### Super User changes
-

#### Admin changes
- Banner text messages can be changed from `anet-dictionary.yml`

`CONNECTION_ERROR_MSG`: Network / Server down message

`VERSION_CHANGED_MSG`: New version message, tell what user should do.

#### System admin changes
- [X] anet.yml or anet-dictionary.yml needs change
- [ ] db needs migration
- [ ] documentation has changed
- [ ] graphql schema has changed

### Checklist
  - [X] Described the user behavior in PR body
  - [X] Referenced/updated all related issues
  - [X] commits follow a `repo#issue: Title` title format and [these 7 rules](https://chris.beams.io/posts/git-commit/)
  - [X] commits have a [clean history](https://epage.github.io/dev/commits/), otherwise PR may be squash-merged
  - [X] Added and/or updated unit tests
  - [ ] Added and/or updated e2e tests
  - [ ] Added and/or updated data migrations
  - [ ] Updated documentation
  - [X] Resolved all build errors and warnings
  - [ ] Opened debt issues for anything not resolved here

### CODE REVIEW

File: client/src/connectionUtils.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import API from "api"
+import { gql } from "apollo-boost"
+import { useEffect } from "react"
+import { Version } from "settings"
+
+const GQL_GET_VERSIN_INFO = gql`
```
Comment (gjvoosten) [inline client/src/connectionUtils.js:6]: Typo in `GQL_GET_VERSION_INFO`

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: client/platform/web-dev/settings.js [hunk 2]
```diff
@@ -3,6 +3,16 @@ import { loadFileAjaxSync } from "../utils"
...
+  query: "{\n  projectVersion\n}\n"
+}
+
+const Version = JSON.parse(
+  loadFileAjaxSync(
+    "/graphql",
+    "application/json",
+    JSON.stringify(VERSION_QUERY)
+  )
+).data.projectVersion
```
Comment (gjvoosten) [inline client/platform/web-dev/settings.js:16]: Suggested change:
```
diff --git a/client/platform/web-dev/settings.js b/client/platform/web-dev/settings.js
index c18221fb8..6ce055b87 100644
--- a/client/platform/web-dev/settings.js
+++ b/client/platform/web-dev/settings.js
@@ -1,17 +1,20 @@
+import { gql } from "apollo-boost"
 import { loadFileAjaxSync } from "../utils"
 
 const Settings = JSON.parse(
   loadFileAjaxSync("/api/admin/dictionary", "application/json")
 )
-const VERSION_QUERY = {
-  query: "{\n  projectVersion\n}\n"
-}
+const GQL_GET_VERSION_INFO = gql`
+  query {
+    projectVersion
+  }
+`
 
 const Version = JSON.parse(
   loadFileAjaxSync(
     "/graphql",
     "application/json",
-    JSON.stringify(VERSION_QUERY)
+    JSON.stringify({ query: GQL_GET_VERSION_INFO })
   )
 ).data.projectVersion
```

File: client/src/api.js [hunk 3]
```diff
@@ -110,6 +110,12 @@ const API = {
     return results
   },
 
+  useApiQueryExtended(query, variables, others) {
+    const results = useQuery(query, { variables, ...others })
+    results.error = results.error && API._handleError(results.error)
+    return results
+  },
+
```
Comment (gjvoosten) [inline client/src/api.js:118]: Suggested change:
```
diff --git a/client/src/api.js b/client/src/api.js
index bd2e622dc..8809a9c04 100644
--- a/client/src/api.js
+++ b/client/src/api.js
@@ -104,13 +104,7 @@ const API = {
       .catch(response => Promise.reject(API._handleError(response)))
   },
 
-  useApiQuery(query, variables) {
-    const results = useQuery(query, { variables })
-    results.error = results.error && API._handleError(results.error)
-    return results
-  },
-
-  useApiQueryExtended(query, variables, others) {
+  useApiQuery(query, variables, others) {
     const results = useQuery(query, { variables, ...others })
     results.error = results.error && API._handleError(results.error)
     return results
```

File: client/src/connectionUtils.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+const GQL_GET_VERSION_INFO = gql`
+  query {
+    projectVersion
+  }
+`
+
+const POLL_INTERVAL_IN_MS = 30_000 // milliseconds
+
+export const useConnectionInfo = () => {
+  const { error, data, stopPolling } = API.useApiQueryExtended(
```
Comment (gjvoosten) [inline client/src/connectionUtils.js:15]: Suggested change:
```
-  const { error, data, stopPolling } = API.useApiQueryExtended(
+  const { error, data, stopPolling } = API.useApiQuery(
```

File: client/src/connectionUtils.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  }, [stopPolling])
+
+  const newVersion = data?.projectVersion
+  const connectionInfo = {
+    // if there is no error and the version doesn't match, we should notify version change
+    newVersion: !error && newVersion !== Version ? newVersion : null,
+    error: !!error
+  }
+
+  return connectionInfo
```
Comment (gjvoosten) [inline client/src/connectionUtils.js:33]: Suggested change:
```
-  const connectionInfo = {
+  return {
     // if there is no error and the version doesn't match, we should notify version change
     newVersion: !error && newVersion !== Version ? newVersion : null,
     error: !!error
   }
-
-  return connectionInfo
```

File: client/src/pages/App.js [hunk 6]
```diff
@@ -161,7 +163,11 @@ const App = ({ pageDispatchers, pageProps }) => {
         appSettings: appState.settings,
         currentUser: appState.currentUser,
         loadAppData: refetch,
-        notifications: appState.notifications
+        notifications: appState.notifications,
+        connection: {
+          error: connectionInfo.error,
+          newVersion: connectionInfo.newVersion
+        }
```
Comment (gjvoosten) [inline client/src/pages/App.js:170]: Suggested change:
```
-        connection: {
-          error: connectionInfo.error,
-          newVersion: connectionInfo.newVersion
-        }
+        connection: { ...connectionInfo }
```

**Review by gjvoosten — CHANGES_REQUESTED** [review]

File: client/platform/web-dev/settings.js (hunk 2 above)
Comment (cemalettin-simsoft) [reply client/platform/web-dev/settings.js]: Removed the `gql`, otherwise the request wasn't being processed by server

**Review by gjvoosten — APPROVED** [review]

**Pull request merged**